src/dbnode/storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -77,24 +72,200 @@ type persistenceFlush struct {
+}
+
+func (i *instrumentation) bootstrapShardsCompleted() {
+	duration := i.nowFn().Sub(i.startShards)
+	i.bootstrapShardsDuration.Record(duration)
+	i.log.Info("bootstrapping shards success", zap.Duration("took", duration))
+}
+
+func (i *instrumentation) persistenceFlushFailed(err error) {
+	i.log.Error("peers bootstrapper bootstrap with persistence flush encountered error",
+		zap.Error(err))
+}
+
+func (i *instrumentation) seriesCheckoutFailed(err error) {
+	i.log.Error("could not checkout series", zap.Error(err))
+}
+
+func (i *instrumentation) seriesLoadFailed(err error) {
+	i.log.Error("could not load series block", zap.Error(err))
+}
+
+func (i *instrumentation) shardBootstrapped(shard uint32, numSeries int64, blockTime time.Time) {
+	i.log.Info("peer bootstrapped shard",
+		zap.Uint32("shard", shard),
+		zap.Int64("numSeries", numSeries),
+		zap.Time("blockStart", blockTime),
+	)
+}
+
+func (i *instrumentation) fetchBootstrapBlocksFailed(err error, shard uint32) {
+	i.log.Error("error fetching bootstrap blocks",
+		zap.Uint32("shard", shard),
+		zap.Error(err),
+	)
+}
+
+func (i *instrumentation) peersBootstrapperIndexForRanges(count int) {
+	i.log.Info("peers bootstrapper bootstrapping index for ranges",
+		zap.Int("shards", count),
+	)
+}
+
+func (i *instrumentation) processingReadersFailed(err error, start time.Time) {
+	i.log.Error("error processing readers", zap.Error(err),
+		zap.Time("timeRange.start", start))
+}
+
+func (i *instrumentation) buildingFileSetIndexSegmentStarted(fields []zapcore.Field) {
+	i.log.Debug("building file set index segment", fields...)
+}
+
+func (i *instrumentation) outOfRetentionIndexSegmentSkipped(fields []zapcore.Field) {
+	i.log.Debug("skipping out of retention index segment", fields...)
+	i.persistedIndexBlocksOutOfRetention.Inc(1)
+}
+
+func (i *instrumentation) buildingInMemoryIndexSegmentStarted(fields []zapcore.Field) {
+	i.log.Info("building in-memory index segment", fields...)
+}
+
+func (i *instrumentation) errorsForRangeEncountered(summaryString string, errorsString []string) {
+	i.log.Info("encountered errors for range",
+		zap.String("requestedRanges", summaryString),
+		zap.Strings("timesWithErrors", errorsString))
+}
+
+func (i *instrumentation) noPeersAvailable(total int, shardIDUint uint32) {
+	i.log.Debug(
+		"0 available peers, unable to peer bootstrap",
+		zap.Int("total", total), zap.Uint32("shard", shardIDUint))
+}
+
+func (i *instrumentation) readConsistencyNotAchieved(
+	bootstrapConsistencyLevel topology.ReadConsistencyLevel,
+	majorityReplicas int,
+	total int,
+	available int,
+) {
+	i.log.Debug(
+		"read consistency not achieved, unable to peer bootstrap",
+		zap.Any("level", bootstrapConsistencyLevel),
+		zap.Int("replicas", majorityReplicas),
+		zap.Int("total", total),
+		zap.Int("available", available))
+}
+
+func (i *instrumentation) peersBootstrapperSourceReadStarted(ctx context.Context) (
+	context.Context, opentracing.Span, bool) {
```

Comment:
Nit (a bit easier on the eyes):
```suggestion
func (i *instrumentation) peersBootstrapperSourceReadStarted(
     ctx context.Context,
) (context.Context, opentracing.Span, bool) {
```